frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx

```diff
@@ -113,25 +110,80 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
     }
   }, [loadingInstances, subscribeToMoreInstances, tenantNamespace, userId]);
 
+  const [dataTemplate, setDataTemplate] = useState<WorkspaceTemplatesQuery>();
+
   const {
-    data: dataTemplate,
     loading: loadingTemplate,
     error: errorTemplate,
-    refetch: refetchTemplate,
+    subscribeToMore: subscribeToMoreTemplates,
   } = useWorkspaceTemplatesQuery({
     variables: { workspaceNamespace },
-    notifyOnNetworkStatusChange: true,
+    onCompleted: setDataTemplate,
   });
 
-  //This useEffect and the reload state are used to simulate the subscription behaviour, it will be removed in the next PR
+  const { instances } = dataInstances?.instanceList ?? {};
+
   useEffect(() => {
-    if (reload && refetchTemplate) {
-      setReload(false);
-      refetchTemplate({ workspaceNamespace });
-    }
-  }, [reload, refetchTemplate, setReload, workspaceNamespace]);
+    if (!loadingTemplate) {
+      subscribeToMoreTemplates({
+        document: updatedWorkspaceTemplates,
+        variables: { workspaceNamespace: `${workspaceNamespace}` },
+        updateQuery: (prev, { subscriptionData }) => {
+          let subscriptionResult: 'modified' | 'added' | 'deleted' | '' = '';
+          const {
+            data,
+          } = subscriptionData as UpdatedWorkspaceTemplatesSubscriptionResult;
+          const newItem = data?.updatedTemplate?.template!;
 
-  const { instances } = dataInstances?.instanceList ?? {};
+          if (!data?.updatedTemplate?.template) return prev;
+
+          if (prev.templateList?.templates) {
+            const oldItem = prev.templateList.templates.find(
+              t =>
+                t?.metadata?.id ===
+                data?.updatedTemplate?.template?.metadata?.id
+            );
+            if (oldItem) {
+              if (JSON.stringify(oldItem) === JSON.stringify(newItem)) {
+                //template have been deleted
+                prev.templateList.templates = prev.templateList?.templates!.filter(
+                  t =>
+                    data?.updatedTemplate?.template?.metadata?.id !==
```

Comment:
```suggestion
                    newItem.metadata?.id !==
```
Also below in another two or three occurrences

Reply:
Done